CONTRIBUTING.md

```diff
@@ -0,0 +1,190 @@
+
+- Workflow
+- Platform
+- Build
+
+This project aims to follow the Kubernetes [Operator pattern](https://kubernetes.io/docs/concepts/extend-kubernetes/operator/)
+
+It uses [Controllers](https://kubernetes.io/docs/concepts/architecture/controller/)
+which provides a reconcile function responsible for synchronizing resources untile the desired state is reached on the cluster
+
+
+# Contributing to the Kogito Serverless Operator
+
+# PreRequisites
+The Operator's controllers and the configurations are generated using the Operator sdk, the code is written in Go, the tasks are executed using a Makefile
+
+In order to build the project, you need to comply with the following requirements:
+
+operator-sdk-v1.24.0 (https://sdk.operatorframework.io/docs/building-operators/golang/installation/)
+
+More information can be found via the [Kubebuilder Documentation](https://book.kubebuilder.io/introduction.html)
+
+Go 1.19 (https://go.dev/dl/)
+
+GNU Make: 
+used to define composite build actions. This should be already installed or available as a package if you have a good OS (https://www.gnu.org/software/make/).
+
+**NOTE:** Run `make help` for more information on all potential `make` targets
+
+
+## Getting Started
+You’ll need a Kubernetes cluster to run against. You can use 
+- [KIND](https://sigs.k8s.io/kind) 
+- [MINIKUBE](https://minikube.sigs.k8s.io)  
+- [CRC](https://console.redhat.com/openshift/create/local) 
+
+to get a local cluster for testing, or run against a remote cluster.
+**Note:** Your controller will automatically use the current context in your kubeconfig file (i.e. whatever cluster `kubectl cluster-info` shows).
+
+### Building
+
+```sh
+make container-build
+```
+
+
+### Modifying the API definitions
+If you are editing the API definitions, generate the manifests such as CRs or CRDs using:
+
+```sh
+make manifests
+```
+
+
+### Running on the cluster
+1. Install Instances of Custom Resources:
+
+```sh
+kubectl apply -f config/samples/
+```
+
+2. Build and push your image to the location specified by `IMG`:
+
+```sh
+make docker-build docker-push IMG=<some-registry>/kogito-serverless-operator:tag
+```
+
+3. Deploy the controller to the cluster with the image specified by `IMG`:
+
+```sh
+make deploy IMG=<some-registry>/kogito-serverless-operator:tag
+```
+
+### Uninstall CRDs
+To delete the CRDs from the cluster:
+
+```sh
+make uninstall
+```
+
+### Undeploy controller
+UnDeploy the controller to the cluster:
+
+```sh
+make undeploy
+```
+
+Minikube or CRC
+
+## Test the Greeting workflow on Minikube
+
+A good starting point to check that everything is working well, it is the [Greeting workflow](https://github.com/kiegroup/kogito-examples/blob/stable/README.md#serverless-workflow-getting-started).
+
+Follow these steps to create a container that you can than deploy as a Service on Kubernetes or KNative.
+
+1. Start Minikube
+```sh 
+minikube start --cpus 4 --memory 4096 --addons registry --insecure-registry "10.0.0.0/24"
```

Comment:
@spolti The insecure registry option is probably meant only to disable SSL communication between the registry and the clients (kaniko or the kube control plane) https://minikube.sigs.k8s.io/docs/handbook/registry/#enabling-insecure-registries in order to avoid the load of the CA certificates into the kaniko image.

When we will have the Platform we will be able to configure also a secure communication but at the moment no.